app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -2559,6 +2562,24 @@ class PluginApi {
   addBulkActionButton(opts) {
     _addBulkButton(opts);
   }
+
+  /**
+   * Allows you add additional search results to the full page search results.
+   * Search results added through this API will be mixed with initial search results
+   * and ranked using the reciprocal ranking fusion algorithm.
+   *
+   * (See `reciprocallyRankedList()` in `discourse/app/lib/search.js` for more
+   * details on how reciprocal ranking works)
+   *
+   *
+   * Example:
+   *
+   * api.addSearchResults(newPostArray);
+   */
+  addSearchResults(results) {
```

Comment:
I'm not sure about this kind of 'push' API. How is the plugin/theme supposed to actually do their ajax request? Or find out what the search term is? They have to connect to a Plugin Outlet on the search page and observe the arguments?

I feel like this API would make more sense as some kind of callback, which can return a promise. Then themes/plugins could do something like:

```js
addSearchResultsCallback(await (term) => {
  const results = await ajax(`my-search-endpoint.json?term=${term}`)
  results.topics;
});
```